File: rb/features.rb
# OpenDataHub SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module OpenDataHubFeatures
  def self.make_feature(name)
    case name
    when "base"
      OpenDataHubBaseFeature.new
    when "ratelimit"
      OpenDataHubRatelimitFeature.new
    when "retry"
      OpenDataHubRetryFeature.new
    when "test"
      OpenDataHubTestFeature.new
    when "timeout"
      OpenDataHubTimeoutFeature.new
    else
      OpenDataHubBaseFeature.new
    end
  end
end

File: py/features.py
# OpenDataHub SDK feature factory

from feature.base_feature import OpenDataHubBaseFeature
from feature.test_feature import OpenDataHubTestFeature


def _make_feature(name):
    features = {
        "base": lambda: OpenDataHubBaseFeature(),
        "test": lambda: OpenDataHubTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
